### DISCUSSION TITLE
Where to find  Endpoint Documentation?

### DISCUSSION THREAD

**Discussion by flomickl** [open-webui/open-webui#4673] [Q&A]
Hi ll,

I have open-webui running with ollama in a container.

Everything is working as expected. I can talk to ollama via open-webui browser and I also can run some basic curl commands to the api from the terminal as well.
But if I try to set up web open-webui with continue in vscode as described here in some way [1]  or try to talk to the endpoint `api/chat`
I got the result 

`405 Method Not Allowed` 

Does anyone have a working example or can tell me what is wrong?
I also couldn't find a documentation where all the API endpoints are listed I potentially could use.

So that is happening with the following request.

request
```bash 
curl -v -X POST --location 'https://example-com/api/chat' \
-H 'Content-Type: application/json' \
-H 'Authorization: Bearer sk-token' \
-d '{
    "model": "llama3.1",
    "prompt": "what is the meaning of life?",
    "stream": false
}';echo
```

response
```bash
{"detail":"Method Not Allowed"}
``` 

Here are the corresponding open-webui logs
```bash
DEBUG [main] Commit session after request
INFO:     10.89.0.10:0 - "POST /api/chat HTTP/1.1" 405 Method Not Allowed
``` 

I choose these request because the vscode extension continue is sending a similar request with the same response.

config.json from continue adjusted from here [1] or [2] and to the continue default config...

```bash
{
  "models": [
    {
      "title": "Oh-My-LLama",
      "provider": "ollama",
      "model": "llama3.1",
      "apiBase": "https://example.com",
      "apiKey": "sk-token"
    }
  ],
  "customCommands": [
    {
      "name": "test",
      "prompt": "{{{ input }}}\n\n What is the meaning of life?.",
      "description": "Something meaningfull"
    }
  ],
  "tabAutocompleteModel": {
    "title": "Oh-My-LLama",
    "provider": "ollama",
    "model": "llama3.1",
    "apiBase": "https://example.com",
    "apiKey": "sk-token"
  }
}
```

[1] https://docs.openwebui.com/tutorial/continue-dev/
[2] https://docs.continue.dev/reference/Model%20Providers/ollama

P.S. That everything is working shows following request example but it is another API endpoint

```bash
curl -v -X POST --location 'https://example-com/ollama/api/generate' \
-H 'Content-Type: application/json' \
-H 'Authorization: Bearer sk-token' \
-d '{
    "model": "llama3.1",
    "prompt": "what is the meaning of life?",
    "stream": false
}';echo
```

open-webUI logs
```bash
INFO  [apps.ollama.main] url: http://ollama:11434
DEBUG [main] Commit session after request
INFO:     10.89.0.10:0 - "POST /ollama/api/generate HTTP/1.1" 200 OK
```

ollama logs
```bash
[GIN] 2024/08/16 - 21:56:55 | 200 |         1m59s |      10.89.6.25 | POST     "/api/generate"
```

And here is my container setup using podman

to start with

```bash 
podman network create ollama-backend
podman volume create ollama_data
podman volume create ollama-webui
```

```bash 
podman run  \
    -it \
    -d \
    --name ollama \
    -p 11434:11434 \
    -v oolama_data:/root/.ollama \
    -e OLLAMA_KEEP_ALIVE=24h \
    -e OLLAMA_HOST=0.0.0.0 \
    --network ollama-backend \
    docker.io/ollama/ollama:latest
``` 

```bash 
podman run  \
    -it \
    -d \
    --name ollama-webui \
    --network ollama-backend \
    -v ollama-webui:/app/backend/data \
    -p 8080:8080 \
    -e OLLAMA_BASE_URLS=http://ollama:11434 \
    -e ENABLE_SIGNUP=true \
    -e WEBUI_NAME="May the FOSS be with you" \
    -e WEBUI_URL=https://example.com \
    -e WEBUI_SECRET_KEY=t0p-s3cr3t \
    -e GLOBAL_LOG_LEVEL="DEBUG" \
    ghcr.io/open-webui/open-webui:v0.3.13
```

**Comment by flomickl** [comment]
any ideas? Is there any documentation of all the available endpoints somewhere?

**Comment by larten** [comment]
@flomickl 
You have to use "apiBase": "http://localhost:3000/ollama" in the continue config.json.

This works with basic docker install.
(docker run -d -p 3000:8080 --gpus=all -v ollama:/root/.ollama -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:ollama)
